common/persistence/sql/sqlplugin/sqlite/plugin.go

```diff
@@ -114,9 +125,37 @@ func (p *plugin) createDBConnection(
 	// Maps struct names in CamelCase to snake without need for db struct tags.
 	db.MapperFunc(strcase.ToSnake)
 
+	// only suitable for in memory databases!
+	if mode := cfg.ConnectAttributes["mode"]; mode == "memory" {
+		// creates temporary DB overlay in order to configure database and schemas
+		err := p.setupSQLiteDatabase(cfg, db)
```

Comment:
This call now conflicts with explicit schema creation in `sqlite_test.go`. Where it should be removed: here or in tests?

Reply:
This is why unit tests are failing on Buildkite.

Reply:
I think it makes sense to handle this inline here to simplify usage of the plugin; I don't know of any cases where it wouldn't make sense to initialize the schema with in-memory mode enabled.

In Temporalite we initialize schema when in-memory mode is being used OR when the db file doesn't already exist. We'd be able to skip the setup for in-memory mode here: https://github.com/DataDog/temporalite/blob/de413afe117f4a5117d1148198622f0f54d5edc2/server.go#L55-L64

As an aside, for file persistence mode I think we should aim for consistency with how migrations work with other sql storage drivers, ie. support sqlite via https://github.com/temporalio/temporal/tree/master/tools/sql.

Reply:
I agree, removing it from tests.